src/main/java/edu/greenblitz/tobyDetermined/subsystems/telescopicArm/Extender.java

```diff
@@ -10,140 +11,160 @@
+			return ExtenderState.OUT_OF_BOUNDS;
+		} else if (lengthInMeters < RobotMap.telescopicArm.extender.MAX_ENTRANCE_LENGTH) {
+			return ExtenderState.IN_WALL_LENGTH;
+		} else if (lengthInMeters < RobotMap.telescopicArm.extender.MAX_LENGTH_IN_ROBOT) {
+			return ExtenderState.IN_ROBOT_BELLY_LENGTH;
+		} else {
+			return ExtenderState.OPEN;
+		}
+	}
+	
+	
+	public static double getFeedForward(double wantedSpeed, double wantedAcceleration, double elbowAngle) {
+		return getStaticFeedForward(elbowAngle) +
+				RobotMap.telescopicArm.extender.kS * Math.signum(wantedSpeed) +
+				RobotMap.telescopicArm.extender.kV * wantedSpeed +
+				RobotMap.telescopicArm.extender.kA * wantedAcceleration;
+	}
+	
+	public static double getStaticFeedForward(double elbowAngle) {
+		return Math.sin(elbowAngle - RobotMap.telescopicArm.elbow.STARTING_ANGLE_RELATIVE_TO_GROUND) * RobotMap.telescopicArm.extender.kG;
+	}
+	
+	private void setLengthByPID(double lengthInMeters) {
+		profiledPIDController.reset(getLength());
+		profiledPIDController.setGoal(lengthInMeters);
+		double feedForward = getFeedForward(
+				profiledPIDController.getSetpoint().velocity, (profiledPIDController.getSetpoint().velocity - motor.getEncoder().getVelocity()) / RoborioUtils.getCurrentRoborioCycle(), Elbow.getInstance().getAngle());
+		SmartDashboard.putNumber("Extender FF", feedForward);
+		motor.getPIDController().setReference(profiledPIDController.getSetpoint().velocity, CANSparkMax.ControlType.kVelocity, 0, feedForward);
+	}
+	
+	public void moveTowardsLength(double lengthInMeters) {
+		// going out of bounds should not be allowed
+		if (getHypotheticalState(lengthInMeters) == ExtenderState.OUT_OF_BOUNDS) {
+			System.err.println("arm Extender is trying to move OUT OF BOUNDS");
+			stop();
+			return;
+		}
+		// arm should not extend to open state when inside the belly (would hit chassis)
+		if (Elbow.getInstance().getState() == Elbow.ElbowState.IN_BELLY && getHypotheticalState(lengthInMeters) == ExtenderState.OPEN) {
+			setLengthByPID(RobotMap.telescopicArm.extender.MAX_ENTRANCE_LENGTH);
+		}else if (Elbow.getInstance().getState() == Elbow.ElbowState.WALL_ZONE && getHypotheticalState(lengthInMeters) != ExtenderState.IN_WALL_LENGTH) {
+			// arm should not extend too much in front of the wall
+			stop();
+		} else {
+			setLengthByPID(lengthInMeters);
+		}
+	}
+	
+	
+	public double getLength() {
+		return motor.getEncoder().getPosition();
+	}
+	
+	public ExtenderState getState() {
+		return state;
+	}
+	
+	/**
+	 * @return the current value of the limit switch
+	 */
+	public boolean getLimitSwitch() {
+		return motor.getReverseLimitSwitch(SparkMaxLimitSwitch.Type.kNormallyOpen).isPressed();
+	}
+	
+	private boolean lastSwitchReading;
+	
+	/**
+	 * @return whether the switch changed status from last periodic this is useful to reset at
+	 */
+	public boolean didSwitchFlip(){
+		return lastSwitchReading ^ getLimitSwitch();
+	}
+	
+	public void resetLength() {
```

Comment:
make reset to receive an angel and default to 0

Reply:
angle?

Reply:
length in Extender, angle in Elbow